pkg/controllers/cloud/configmap_controller.go

```diff
@@ -54,43 +81,78 @@ func (r *ConfigMapReconciler) Reconcile(_ context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	// set the cloud resource prefix.
 	if err := r.setCloudResourcePrefix(newControllerConfig.CloudResourcePrefix); err != nil {
 		// Unable to continue as prefix is not valid.
 		r.Log.Error(err, "exiting nephe-controller")
 		os.Exit(1)
 	}
+
+	// set the cloud sync interval.
+	r.setCloudSyncInterval(newControllerConfig.CloudSyncInterval)
+
 	return ctrl.Result{}, nil
 }
 
 // setCloudResourcePrefix sets the CloudResourcePrefix, returns error when invalid.
 func (r *ConfigMapReconciler) setCloudResourcePrefix(prefix string) error {
 	if len(prefix) == 0 {
-		r.Log.Info("Setting CloudResourcePrefix to default", "Old", r.ControllerConfig.CloudResourcePrefix,
-			"New", config.DefaultCloudResourcePrefix)
+		r.Log.Info("Setting CloudResourcePrefix to default", "CloudResourcePrefix", config.DefaultCloudResourcePrefix)
 		r.ControllerConfig.CloudResourcePrefix = config.DefaultCloudResourcePrefix
 		return nil
 	}
 
 	if !(config.ValidateName(prefix)) {
 		return fmt.Errorf("invalid CloudResourcePrefix %s, "+
-			"First and last characters should be alphanumeric and "+
-			"'-' characters are allowed only in the middle", prefix)
+			"Only alphanumeric and '-' characters are allowed."+
+			" Special character '-' is only allowed at the middle", prefix)
 	}
 
-	// Update the CloudResourcePrefix
+	// Update the CloudResourcePrefix.
 	if r.ControllerConfig.CloudResourcePrefix != prefix {
-		r.Log.V(1).Info("Updating the CloudResourcePrefix",
-			"Old", r.ControllerConfig.CloudResourcePrefix, "New", prefix)
+		r.Log.V(1).Info("Updating the CloudResourcePrefix", "CloudResourcePrefix", prefix)
 		r.ControllerConfig.CloudResourcePrefix = prefix
 	}
 	return nil
 }
 
+// setCloudSyncInterval sets the CloudSyncInterval.
+func (r *ConfigMapReconciler) setCloudSyncInterval(interval int64) {
+	if interval == 0 {
+		r.Log.Info("Setting CloudSyncInterval to default", "CloudSyncInterval", config.DefaultCloudSyncInterval)
+		r.ControllerConfig.CloudSyncInterval = config.DefaultCloudSyncInterval
+	} else if interval < config.DefaultCloudSyncInterval {
+		r.ControllerConfig.CloudSyncInterval = config.DefaultCloudSyncInterval
+		r.Log.Info(fmt.Sprintf("Invalid CloudSyncInterval %d"+
+			"CloudSyncInterval should be >= %d seconds. So using default interval", interval, config.DefaultCloudSyncInterval))
+	} else if r.ControllerConfig.CloudSyncInterval != interval {
```

Comment:
nit: for readability. keep it same as in other comparison.
```suggestion
	} else if interval != r.ControllerConfig.CloudSyncInterval {
```

Reply:
Solved.